src/dbnode/storage/index/postings_list_cache.go

```diff
@@ -220,29 +232,42 @@ func (q *PostingsListCache) Report() {
 }
 
 func (q *PostingsListCache) emitCacheGetMetrics(patternType PatternType, hit bool) {
-	switch {
-	case patternType == PatternTypeRegexp && hit:
-		q.metrics.regexp.hits.Inc(1)
-	case patternType == PatternTypeRegexp && !hit:
-		q.metrics.regexp.misses.Inc(1)
-	case patternType == PatternTypeTerm && hit:
-		q.metrics.term.hits.Inc(1)
-	case patternType == PatternTypeTerm && !hit:
-		q.metrics.term.misses.Inc(1)
+	var method *postingsListCacheMethodMetrics
+	switch patternType {
+	case PatternTypeRegexp:
+		method = q.metrics.regexp
+	case PatternTypeTerm:
+		method = q.metrics.term
+	case PatternTypeField:
+		method = q.metrics.field
+	default:
+		method = q.metrics.unknown
+	}
+	if hit {
+		method.hits.Inc(1)
+	} else {
+		method.misses.Inc(1)
 	}
 }
 
 func (q *PostingsListCache) emitCachePutMetrics(patternType PatternType) {
-	if patternType == PatternTypeRegexp {
+	switch patternType {
+	case PatternTypeRegexp:
 		q.metrics.regexp.puts.Inc(1)
-	} else {
+	case PatternTypeTerm:
 		q.metrics.term.puts.Inc(1)
+	case PatternTypeField:
+		q.metrics.field.puts.Inc(1)
+	default:
+		q.metrics.unknown.puts.Inc(1)
```

Comment:
super nit: Can you add a "should never happen" under the default case so it doesnt seem like thats something that might happen regularly

Reply:
sure thing

Reply:
done